test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -817,7 +817,26 @@ protected Set<String> preserveILMPolicyIds() {
             ".fleet-file-tohost-meta-ilm-policy",
             ".deprecation-indexing-ilm-policy",
             ".monitoring-8-ilm-policy",
-            "behavioral_analytics-events-default_policy"
+            "behavioral_analytics-events-default_policy",
+            "logs-apm.app_logs-default_policy",
+            "logs-apm.error_logs-default_policy",
+            "metrics-apm.app_metrics-default_policy",
+            "metrics-apm.internal_metrics-default_policy",
+            "metrics-apm.service_destination_10m_metrics-default_policy",
+            "metrics-apm.service_destination_1m_metrics-default_policy",
+            "metrics-apm.service_destination_60m_metrics-default_policy",
+            "metrics-apm.service_summary_10m_metrics-default_policy",
+            "metrics-apm.service_summary_1m_metrics-default_policy",
+            "metrics-apm.service_summary_60m_metrics-default_policy",
+            "metrics-apm.service_transaction_10m_metrics-default_policy",
+            "metrics-apm.service_transaction_1m_metrics-default_policy",
+            "metrics-apm.service_transaction_60m_metrics-default_policy",
+            "metrics-apm.transaction_10m_metrics-default_policy",
+            "metrics-apm.transaction_1m_metrics-default_policy",
+            "metrics-apm.transaction_60m_metrics-default_policy",
+            "traces-apm.rum_traces-default_policy",
+            "traces-apm.sampled_traces-default_policy",
+            "traces-apm.traces-default_policy"
```

Comment:
Do you think all these policies are necessary? Do you seethe retention of the different data streams diverging a lot individually?

If the assumption that these default retention per data stream will not be diverging from each other a lot, I think you should consider merging them on their retention instead of having them separately.

I am saying this because maintaining 18 policies can be a lot; from what I see they mainly configure retention (if we consider rollover an implementation detail):
- 3 policies have retention `10d`
- 7 policies have retention `90d`
- 4 policies have retention `180d`
- 4 policies have retention `390d`

These could be easily simplified to 4 policies, which would also mean 4 component templates. Wouldn't that be easier to maintain?

Reply:
> Do you think all these policies are necessary? Do you seethe retention of the different data streams diverging a lot individually?

Good point, I considered this when creating this PR but these are also taken from defaults, and changing them might have unintended side-effects which we are trying to avoid. Technically, I don't see anything that would break by changing the policy names and consolidating the default policies but I have been proven wrong before. Additionally, it will also mean that we should be prepared to add more ILM policies if we in future want to change the policy configuration for specific datastreams. @axw wdyt?

Reply:
@lahsivjar The policies simply match those installed by the APM integration before introduction of the apm-data plugin.  I think that's the right goal and we should not try to take this opportunity to change too much at once.